lib/tasks/compact_index.rake

```diff
@@ -1,18 +1,41 @@
 namespace :compact_index do
   desc "Fill yanked_at with current time"
-  task migrate_yanked_at: :environment do
-    Version.where(yanked_at: nil).find_in_batches do |versions|
-      versions.each { |v| v.update_attribute :yanked_at, Time.now.utc }
+  task backfill_yanked_at: :environment do
+    without_yanked_at = Version.where(yanked_at: nil)
+    mod = ENV['shard']
+    without_yanked_at = without_yanked_at.where("id % 4 = ?", mod.to_i) if mod
+
+    total = without_yanked_at.count
+    i = 0
+    puts "Total: #{total}"
+    without_yanked_at.find_each do |version|
+      version.update_attribute :yanked_at, Time.now.utc
```

Comment:
>  better yet remove this column and grab the correct information from the deletions table.

`deletions` doesn't store version id. To get the time when a gem was yanked, one would need to join three tables. It will further slow down `versions` response which is already at 10s.